homeassistant/components/sensor/tautulli.py

```diff
@@ -0,0 +1,153 @@
+"""
+A platform which allows you to get information from Tautulli.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/components/sensor.tautulli/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_API_KEY, CONF_HOST,
+                                 CONF_MONITORED_CONDITIONS, CONF_NAME,
+                                 CONF_PORT, CONF_SSL, CONF_VERIFY_SSL)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['pytautulli==0.4.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MONITORED_USERS = 'monitored_users'
+CONF_UPDATE_INTERVAL = 'update_interval'
+
+DEFAULT_NAME = 'Tautulli'
+DEFAULT_PORT = '8181'
+DEFAULT_CONDITIONS = 'None'
+DEFAULT_USERS = 'None'
+DEFAULT_SSL = False
+DEFAULT_VERIFY_SSL = True
+
+TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
+    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=DEFAULT_CONDITIONS):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_MONITORED_USERS, default=DEFAULT_USERS):
+        vol.All(cv.ensure_list, [cv.string]),
+    })
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create the sensor."""
+    from pytautulli import Tautulli
+
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    api_key = config.get(CONF_API_KEY)
+    monitored_conditions = config.get(CONF_MONITORED_CONDITIONS)
+    user = config.get(CONF_MONITORED_USERS)
+    use_ssl = config.get(CONF_SSL)
+    verify_ssl = config.get(CONF_VERIFY_SSL)
+
+    session = async_get_clientsession(hass, verify_ssl)
+    tautulli = TautulliData(Tautulli(
+        host, port, api_key, hass.loop, session, use_ssl))
+
+    await tautulli.test_connection()
+
+    if not tautulli.test_connection:
```

Comment:
This is wrong. We need to check the return value from `await tautulli.test_connection()`.